src/Csv/CsvBuilder.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace App\Csv;
+
+use DateTime;
+
+class CsvBuilder
+{
+    public function buildCsv()
+    {
+        return $this->arrayToCsv($this->buildPayrollArray(), 'test11.csv');
+    }
+
+    private function buildPayrollArray() {
+        $format = 'd-m-Y';
+        $currentDate = New DateTime;
+        $currentMonth = (int) $currentDate->format('n');
+        $remainingMonths = [];
+
+        for ($month = $currentMonth; $month <= 12; $month++) {
+            $monthName = date('F', mktime(0, 0, 0, $month, 1));
+            $bonusDate = $this->determineBonusDate($month);
+            $salaryDate = $this->determineSalaryDate($month);
+
+            $remainingMonths[] = [
+                'Month' => $monthName,
+                'Bonusdate' => $bonusDate->format($format),
+                'Salarydate' => $salaryDate->format($format)
+            ];
+        }
+
+        return $remainingMonths;
+    }
+
+    private function arrayToCsv(array $payrollArray, string $csvName): string
+    {
+        $filePath = sys_get_temp_dir() . '/' . $csvName;
+        $file = fopen($filePath, 'w');
+
+        fputcsv($file, array_keys($payrollArray[0]));
+
+        foreach ($payrollArray as $row) {
+            fputcsv($file, $row);
+        }
+
+        fclose($file);
+
+        return $filePath;
+    }
+
+    private function determineBonusDate(string $month)
+    {
+        $bonusDate = date('d-m-Y', mktime(0, 0, 0, $month, 15));
+        $bonusDateTime = new DateTime($bonusDate);
+
+        switch ($bonusDateTime) {
```

Comment:
Als je alleen true/false hebt is een if statement denk ik makkelijker. Buiten true/false kan je niet meer cases hebben in een switch, dus dan heeft het niet veel zin